docs/getting-started/delete-on-dockstore.rst

```diff
@@ -0,0 +1,44 @@
+Deleting an entry on Dockstore
```

Comment:
Nit: entry -> Entry because the other side bar titles are capitalized

![image](https://github.com/dockstore/dockstore-documentation/assets/25287123/fb3ecea8-8292-4236-bbe3-f226255253fe)